examples/echo/main.c

```diff
@@ -143,16 +144,35 @@ static void handle_rx (YggWorker   *worker,
   g_bytes_unref (data);
 }
 
+static gboolean
+sigint_callback(gpointer main_loop) {
+  // Hint: this is good place to perform any resource clean up
+  // before the application terminates.
+  g_debug ("handling signal SIGINT");
+  // Stops main loop
+  g_main_loop_quit(main_loop);
+  return G_SOURCE_REMOVE;
+}
+
 gint
 main (gint   argc,
       gchar *argv[])
 {
   g_set_prgname ("yggdrasil-worker-echo");
 
+  // Create main context for main loop
+  GMainContext *main_context = g_main_context_default ();
+  g_autoptr (GMainLoop) loop = g_main_loop_new (main_context, FALSE);
+
+  // Attach signal sources to the main_context
+  g_autoptr (GSource) sigint_source = g_unix_signal_source_new (SIGINT);
+  g_source_attach (sigint_source, main_context);
+  g_source_set_callback (sigint_source, G_SOURCE_FUNC(sigint_callback), loop, NULL);
```

Comment:
I think this needs a space too.
```suggestion
  g_source_set_callback (sigint_source, G_SOURCE_FUNC (sigint_callback), loop, NULL);
```